llvm/lib/Target/AMDGPU/AMDGPULaneMaskUtils.h

```diff
@@ -0,0 +1,109 @@
+//===- AMDGPULaneMaskUtils.h - Exec/lane mask helper functions -*- C++ -*--===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIB_TARGET_AMDGPU_UTILS_AMDGPULANEMASKUTILS_H
+#define LLVM_LIB_TARGET_AMDGPU_UTILS_AMDGPULANEMASKUTILS_H
+
+#include "GCNSubtarget.h"
+#include "llvm/CodeGen/Register.h"
+
+namespace llvm {
+
+class GCNSubtarget;
+
+namespace AMDGPU {
+
+struct LaneMaskConstants {
+  Register ExecReg;
+  Register VccReg;
+  unsigned AndOpc;
+  unsigned AndTermOpc;
+  unsigned AndN2Opc;
+  unsigned AndN2SaveExecOpc;
+  unsigned AndN2TermOpc;
+  unsigned AndSaveExecOpc;
+  unsigned AndSaveExecTermOpc;
+  unsigned BfmOpc;
+  unsigned CMovOpc;
+  unsigned CSelectOpc;
+  unsigned MovOpc;
+  unsigned MovTermOpc;
+  unsigned OrOpc;
+  unsigned OrTermOpc;
+  unsigned OrSaveExecOpc;
+  unsigned XorOpc;
+  unsigned XorTermOpc;
+  unsigned WQMOpc;
+
+  static constexpr LaneMaskConstants getWave32() {
```

Comment:
I meant something more like:

```
  static constexpr LaneMaskConstants Wave32 = {}
  static constexpr LaneMaskConstants Wave64 = {};

static const LaneMaskConstants &get(const GCNSubtarget &ST) {
  return isWave32() ? Wave32 : Wave64;
}
```

You could also push this the setting into a proper constructor with a flag so you have the name op the opcode with the value

Reply:
or 

```
template <bool IsWave32>
LaneMaskConstants:() : XorOpc(IsWave32 ? ...), ... 
```

Reply:
Thanks, that makes sense!
I've made changes based no your suggests.